spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -1351,6 +1378,15 @@ protected String getRemoteFilename(String remoteFilePath) {
 		}
 	}
 
+	private FileExistsMode resolveFileExistsMode(Message<?> message) {
+		if (this.fileExistsModeExpression != null) {
+			EvaluationContext evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
+			evaluationContext.setVariable("fileExistsMode", this.fileExistsMode);
```

Comment:
Why do we need this variable in the expression?
In most cases no one is going to care about whatever is there by default in the gateway setting.
And with that we could use a global `EvaluationContext` created only once in the `doInit()`.

Reply:
I've removed the unnecessary variable and implemented a global `EvaluationContext` that's created once in `doInit()`. 

Additionally, I was thinking that in the long term, it might be beneficial to move the Expression evaluation logic to the `IntegrationObjectSupport` class. This could standardize expression evaluation across Spring Integration and reduce duplicate code in each component. Would this be something worth considering?

Reply:
Feel free to raise a GH issue with more info or even draft some PoC.
And we will go from there this or other way.